Gems/Atom/Tools/ShaderManagementConsole/Scripts/GenerateShaderVariantListForAllShaders.py

```diff
@@ -0,0 +1,44 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+import re
+import json
+import GenerateShaderVariantListUtil
+
+def search_and_save_shadervariant(searchPath):
+    for subdir, dirs, files in os.walk(searchPath):
+        for file in files:
+            filename = os.path.join(subdir, file)
+            if re.search('^.*\.shader$', filename):
+                shaderVariantList, defaultShaderVariantListFilePath = GenerateShaderVariantListUtil.create_shadervariantlist_for_shader(filename)
+                if shaderVariantList.shaderVariants:
+                    azlmbr.shader.SaveShaderVariantListSourceData(defaultShaderVariantListFilePath, shaderVariantList)
+
+
+def main():
+    print("==== Begin processing shader ==========================================================")
+    # Search all .shader in Engine Gems folder
+    search_and_save_shadervariant(azlmbr.paths.engroot + '/Gems')
+
+    # Search in project folder
+    search_and_save_shadervariant(azlmbr.paths.projectroot + '/Shaders')
+    search_and_save_shadervariant(azlmbr.paths.projectroot + '/Assets')
```

Comment:
I believe the engine now supports having externally registered gems, outside the engine folder and outside the project folder. So it's not enough to just scan these three folders. A more reliable way to find all the shaders would be to call EnumerateAssets. I'm not sure whether that's exposed to python or not though. It looks like it would need to be added in AssetManagerComponent::Reflect https://github.com/o3de/o3de/blob/f4bda1feee338b0426f8ee7783cc5d903efc5b43/Code/Framework/AzCore/AzCore/Asset/AssetManagerComponent.cpp#L198

Reply:
I try to add `AssetCatalogRequests::EnumerateAssets` to BehaviorContext, but I got a warning
```
BehaviorMethod EnumerateAssets: Parameter at [0] index expects (AZStd::function<void>:{73369FC8-B59B-5B48-B61A-ECCBF94031C4}) for method but got type (<class 'function'>)
```
Looks like Python function didn't get translated to `AZStd::function<void>` correctly.
Do you know the right way to use it?

Reply:
I am able to call the function in c++ and pass back the list to python.